### ISSUE TITLE
Compiling pyspotify 1 on Windows

### ISSUE THREAD

**Issue by DrFunk** [mopidy/pyspotify#63]
Hi!

I tried compiling on WinXP and ran into several problems I thought you'd like to know.
First, compilation does not work for me with the latest version of gcc (the option '-mno-cygwin' is no longer supported).
Then I tried to downgrade to gcc 3.4.5 but I got compilation errors (log below).
Apparently PyObject_HEAD is not recognized, although we do have #include <Python.h> at the beginning of module.c
Do you have any idea on how to fix this or is there something stupid I'm missing?

Thanks in advance!

Compilation log:

```
~\mopidy-pyspotify-3d62577>python setup.py build
running build
running build_py
creating build
creating build\lib.win32-2.7
creating build\lib.win32-2.7\spotify
copying spotify\__init__.py -> build\lib.win32-2.7\spotify
creating build\lib.win32-2.7\spotify\audiosink
copying spotify\audiosink\alsa.py -> build\lib.win32-2.7\spotify\audiosink
copying spotify\audiosink\gstreamer.py -> build\lib.win32-2.7\spotify\audiosink
copying spotify\audiosink\oss.py -> build\lib.win32-2.7\spotify\audiosink
copying spotify\audiosink\portaudio.py -> build\lib.win32-2.7\spotify\audiosink
copying spotify\audiosink\__init__.py -> build\lib.win32-2.7\spotify\audiosink
creating build\lib.win32-2.7\spotify\manager
copying spotify\manager\container.py -> build\lib.win32-2.7\spotify\manager
copying spotify\manager\playlist.py -> build\lib.win32-2.7\spotify\manager
copying spotify\manager\session.py -> build\lib.win32-2.7\spotify\manager
copying spotify\manager\__init__.py -> build\lib.win32-2.7\spotify\manager
running build_ext
building 'spotify._spotify' extension
creating build\temp.win32-2.7
creating build\temp.win32-2.7\Release
creating build\temp.win32-2.7\Release\src
C:\MinGW\bin\gcc.exe -mno-cygwin -mdll -O -Wall -Isrc -IC:\Python27\include -IC:\Python27\PC -c src/module.c -o build\temp.win32-2.7\Release\src\module.o
src/module.c:3:28: libspotify/api.h: No such file or directory
In file included from src/module.c:5:
src/artist.h:2: error: syntax error before "sp_artist"
src/artist.h:2: warning: no semicolon at end of struct or union
src/artist.h:3: warning: type defaults to `int' in declaration of `Artist'
src/artist.h:3: warning: data definition has no type or storage class
src/artist.h:9: error: syntax error before '*' token
In file included from src/module.c:6:
src/artistbrowser.h:5: error: syntax error before "sp_artistbrowse"
src/artistbrowser.h:5: warning: no semicolon at end of struct or union
src/artistbrowser.h:6: warning: type defaults to `int' in declaration of `ArtistBrowser'
src/artistbrowser.h:6: warning: data definition has no type or storage class
src/artistbrowser.h:13: error: syntax error before '*' token
In file included from src/module.c:7:
src/album.h:2: error: syntax error before "sp_album"
src/album.h:2: warning: no semicolon at end of struct or union
src/album.h:3: warning: type defaults to `int' in declaration of `Album'
src/album.h:3: warning: data definition has no type or storage class
src/album.h:9: error: syntax error before '*' token
In file included from src/module.c:8:
src/albumbrowser.h:4: error: syntax error before "sp_albumbrowse"
src/albumbrowser.h:4: warning: no semicolon at end of struct or union
src/albumbrowser.h:6: error: syntax error before '}' token
src/albumbrowser.h:6: warning: type defaults to `int' in declaration of `AlbumBrowser'
src/albumbrowser.h:6: warning: data definition has no type or storage class
src/albumbrowser.h:13: error: syntax error before '*' token
In file included from src/module.c:9:
src/link.h:2: error: syntax error before "sp_link"
src/link.h:2: warning: no semicolon at end of struct or union
src/link.h:3: warning: type defaults to `int' in declaration of `Link'
src/link.h:3: warning: data definition has no type or storage class
In file included from src/module.c:10:
src/playlist.h:5: error: syntax error before "sp_playlist"
src/playlist.h:5: warning: no semicolon at end of struct or union
src/playlist.h:6: warning: type defaults to `int' in declaration of `Playlist'
src/playlist.h:6: warning: data definition has no type or storage class
src/playlist.h:14: error: syntax error before "sp_playlist_callbacks"
src/playlist.h:14: warning: no semicolon at end of struct or union
src/playlist.h:17: error: syntax error before '}' token
src/playlist.h:17: warning: type defaults to `int' in declaration of `playlist_callback'
src/playlist.h:17: warning: data definition has no type or storage class
src/playlist.h:21: error: syntax error before "sp_playlist"
src/playlist.h:21: warning: no semicolon at end of struct or union
src/playlist.h:22: warning: type defaults to `int' in declaration of `callbacks'
src/playlist.h:22: warning: data definition has no type or storage class
src/playlist.h:23: error: conflicting types for 'next'
src/playlist.h:16: error: previous declaration of 'next' was here
src/playlist.h:24: error: syntax error before '}' token
src/playlist.h:24: warning: type defaults to `int' in declaration of `pl_cb_entry'
src/playlist.h:24: warning: data definition has no type or storage class
src/playlist.h:26: error: syntax error before '*' token
In file included from src/module.c:11:
src/playlistcontainer.h:5: error: syntax error before "sp_playlistcontainer"
src/playlistcontainer.h:5: warning: no semicolon at end of struct or union
src/playlistcontainer.h:6: warning: type defaults to `int' in declaration of `PlaylistContainer'
src/playlistcontainer.h:6: warning: data definition has no type or storage class
src/playlistcontainer.h:12: error: syntax error before '*' token
src/playlistcontainer.h:16: error: syntax error before "sp_playlistcontainer_callbacks"
src/playlistcontainer.h:16: warning: no semicolon at end of struct or union
src/playlistcontainer.h:18: error: conflicting types for 'next'
src/playlist.h:23: error: previous declaration of 'next' was here
src/playlistcontainer.h:18: error: conflicting types for 'next'
src/playlist.h:23: error: previous declaration of 'next' was here
src/playlistcontainer.h:19: error: syntax error before '}' token
src/playlistcontainer.h:19: warning: type defaults to `int' in declaration of `playlistcontainer_callback'
src/playlistcontainer.h:19: warning: data definition has no type or storage class
src/playlistcontainer.h:23: error: syntax error before "sp_playlistcontainer"
src/playlistcontainer.h:23: warning: no semicolon at end of struct or union
src/playlistcontainer.h:24: warning: type defaults to `int' in declaration of `callbacks'
src/playlistcontainer.h:24: warning: data definition has no type or storage class
src/playlistcontainer.h:25: error: conflicting types for 'next'
src/playlistcontainer.h:18: error: previous declaration of 'next' was here
src/playlistcontainer.h:25: error: conflicting types for 'next'
src/playlist.h:23: error: previous declaration of 'next' was here
src/playlistcontainer.h:26: error: syntax error before '}' token
src/playlistcontainer.h:26: warning: type defaults to `int' in declaration of `plc_cb_entry'
src/playlistcontainer.h:26: warning: data definition has no type or storage class
In file included from src/module.c:12:
src/playlistfolder.h:7: error: syntax error before "sp_uint64"
src/playlistfolder.h:7: warning: no semicolon at end of struct or union
src/playlistfolder.h:8: warning: type defaults to `int' in declaration of `_type'
src/playlistfolder.h:8: warning: data definition has no type or storage class
src/playlistfolder.h:9: warning: type defaults to `int' in declaration of `PlaylistFolder'
src/playlistfolder.h:9: warning: data definition has no type or storage class
src/playlistfolder.h:15: error: syntax error before '*' token
In file included from src/module.c:13:
src/search.h:2: error: syntax error before "sp_search"
src/search.h:2: warning: no semicolon at end of struct or union
src/search.h:3: warning: type defaults to `int' in declaration of `Results'
src/search.h:3: warning: data definition has no type or storage class
src/search.h:9: error: syntax error before '*' token
In file included from src/module.c:14:
src/session.h:3: error: syntax error before "sp_session"
src/session.h:3: warning: no semicolon at end of struct or union
src/session.h:5: error: syntax error before '}' token
src/session.h:5: warning: type defaults to `int' in declaration of `Session'
src/session.h:5: warning: data definition has no type or storage class
src/session.h:7: error: syntax error before '*' token
src/session.h:7: warning: type defaults to `int' in declaration of `g_session'
src/session.h:7: warning: data definition has no type or storage class
src/session.h:16: error: syntax error before '*' token
In file included from src/module.c:15:
src/toplistbrowser.h:3: error: syntax error before "sp_toplistbrowse"
src/toplistbrowser.h:3: warning: no semicolon at end of struct or union
src/toplistbrowser.h:4: warning: type defaults to `int' in declaration of `ToplistBrowser'
src/toplistbrowser.h:4: warning: data definition has no type or storage class
src/toplistbrowser.h:10: error: syntax error before '*' token
In file included from src/module.c:16:
src/track.h:2: error: syntax error before "sp_track"
src/track.h:2: warning: no semicolon at end of struct or union
src/track.h:3: warning: type defaults to `int' in declaration of `Track'
src/track.h:3: warning: data definition has no type or storage class
src/track.h:9: error: syntax error before '*' token
In file included from src/module.c:17:
src/image.h:2: error: syntax error before "sp_image"
src/image.h:2: warning: no semicolon at end of struct or union
src/image.h:3: warning: type defaults to `int' in declaration of `Image'
src/image.h:3: warning: data definition has no type or storage class
src/image.h:9: error: syntax error before '*' token
In file included from src/module.c:18:
src/user.h:3: error: syntax error before "sp_user"
src/user.h:3: warning: no semicolon at end of struct or union
src/user.h:4: warning: type defaults to `int' in declaration of `User'
src/user.h:4: warning: data definition has no type or storage class
src/user.h:10: error: syntax error before '*' token
src/module.c: In function `init_spotify':
src/module.c:70: error: `SPOTIFY_API_VERSION' undeclared (first use in this function)
src/module.c:70: error: (Each undeclared identifier is reported only once
src/module.c:70: error: for each function it appears in.)
error: command 'gcc' failed with exit status 1
```

**Comment by jodal** [comment]
Due to the imminent release of the 2.0 rewrite of pyspotify, compilation of pyspotify 1.x on Windows will probably never be documented/supported. I'm closing this issue.

Feel free to open a new issue to discuss how to compile pyspotify 2 on Windows.
